nifi-commons/nifi-security-utils/src/test/groovy/org/apache/nifi/security/util/CertificateUtilsTest.groovy

```diff
@@ -272,4 +275,179 @@ class CertificateUtilsTest extends GroovyTestCase {
         assert convertedCertificate instanceof X509Certificate
         assert convertedCertificate == EXPECTED_NEW_CERTIFICATE
     }
+
+    @Test
+    void testShouldDetermineClientAuthStatusFromSocket() {
+        // Arrange
+        SSLSocket needSocket = [getNeedClientAuth: { -> true }] as SSLSocket
```

Comment:
I'm seeing some Java 8 stuff (Lambdas, e.g) but this PR is against 0.x, which needs Java 7 language target